File: code.exs
defmodule Benchmark do
  @iterations 1_000_000

  def run do
    start_time = :os.system_time(:microsecond)

    sum =
      Enum.reduce(0..(@iterations - 1), 0, fn i, acc ->
        Enum.reduce(0..99, acc, fn j, inner_acc ->
          inner_acc + i * j
        end)
      end)

    end_time = :os.system_time(:microsecond)
    duration = (end_time - start_time) / 1_000_000

    IO.puts("Elixir Result: #{sum}")
    IO.puts("Elixir Time taken: #{duration} seconds")
  end
end

Benchmark.run()

File: code.rs
use std::time::Instant;

const ITERATIONS: i64 = 1_000_000;

fn main() {
    let start = Instant::now();

    let mut sum: i64 = 0;
    for i in 0..ITERATIONS {
        for j in 0..100 {
            sum += i * j;
        }
    }

    let duration = start.elapsed();
    println!("Rust Result: {}", sum);
    println!("Rust Time taken: {:?} seconds", duration.as_secs_f64());
}
